nixos/modules/services/networking/zeronsd.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.zeronsd;
+in {
+  options.services.zeronsd.enable = mkEnableOption (lib.mdDoc "ZeroNSD");
+
+  options.services.zeronsd.servedNetworks = mkOption {
+    default = [];
+    example = [ "a8a2c3c10c1a68de" ];
+    type = types.listOf types.str;
+    description = lib.mdDoc "List of ZeroTier Network IDs to start zeronsd instance for.";
+  };
+
+  options.services.zeronsd.domain = mkOption {
+    default = "home.arpa";
+    type = types.str;
+    description = lib.mdDoc ''
+      Domain under which zerotier records will be available
+    '';
+  };
+
+  options.services.zeronsd.apiTokenFile = mkOption {
+    default = null;
+    type = types.path;
+    description = lib.mdDoc ''
+      Path to a file containing the API Token for ZeroTier Central
+    '';
+  };
+
+  options.services.zeronsd.logLevel = mkOption {
+    default = "info";
+    type = types.str;
+    description = lib.mdDoc ''
+      Log Level (one of off, error, warn, info, debug, trace)
+    '';
+  };
+
+  options.services.zeronsd.wildcardMode = mkOption {
+    default = false;
+    type = types.bool;
+    description = lib.mdDoc ''
+      Enables wildcard mode, where all member names get a wildcard in this format: `*.<name>.<tld>`; this points at the member's IP address(es).
+    '';
+  };
+
+  options.services.zeronsd.package = mkOption {
+    default = pkgs.zeronsd;
+    defaultText = literalExpression "pkgs.zeronsd";
+    type = types.package;
+    description = lib.mdDoc ''
+      ZeroNSD package to use.
+    '';
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services = builtins.listToAttrs (builtins.map (net: {
```

Comment:
why doesn't this just use `lib.mapListToAttrs`?

Reply:
Hmmm, I don't have mapListToAttrs, I can only find it in https://github.com/NixOS/nixpkgs/blob/7f2ec624dda8368c0ffaee325531bf82dd714632/pkgs/top-level/release-alternatives.nix#L50 and this does not seem to be exported anywhere.